app/controllers/users.js

```diff
@@ -36,153 +47,192 @@ exports.signup = function(req, res) {
   if (req.user && req.user._id && req.body.avatar !== undefined &&
     /\d/.test(req.body.avatar) && avatars[req.body.avatar]) {
     User.findOne({
       _id: req.user._id
     })
-    .exec(function(err, user) {
-      user.avatar = avatars[req.body.avatar];
-      user.save();
-    });
+      .exec((err, user) => {
+        user.avatar = avatars[req.body.avatar];
+        user.save();
+      });
   }
   return res.redirect('/#!/app');
 };
 
-exports.addDonation = function(req, res) {
+exports.addDonation = (req, res) => {
   if (req.body && req.user && req.user._id) {
     // Verify that the object contains crowdrise data
-    if (req.body.amount && req.body.crowdrise_donation_id && req.body.donor_name) {
+    if (req.body.amount &&
+      req.body.crowdrise_donation_id &&
+      req.body.donor_name) {
       User.findOne({
         _id: req.user._id
       })
-      .exec(function(err, user) {
+        .exec((err, user) => {
         // Confirm that this object hasn't already been entered
-        var duplicate = false;
-        for (var i = 0; i < user.donations.length; i++ ) {
-          if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
-            duplicate = true;
+          let duplicate = false;
+          for (let i = 0; i < user.donations.length; i += 1) {
+            if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
+              duplicate = true;
+            }
           }
-        }
-        if (!duplicate) {
-          console.log('Validated donation');
-          user.donations.push(req.body);
-          user.premium = 1;
-          user.save();
-        }
-      });
+          if (!duplicate) {
+            user.donations.push(req.body);
+            user.premium = 1;
+            user.save();
+          }
+        });
     }
   }
   res.send();
 };
 
 /**
  *  Show profile
+ * @param {object} req - the request object
+ * @param {object} res - the response object
+ * @returns {func} Shows the user profile
  */
-exports.show = function(req, res) {
-  var user = req.profile;
+exports.show = (req, res) => {
+  const user = req.profile;
 
   res.render('users/show', {
     title: user.name,
-    user: user
+    user
   });
 };
 
 /**
  * Send User
+ * @param {object} req - the request object
+ * @param {object} res - the response object
+ * @returns {func} Returns the user details back
  */
-exports.me = function(req, res) {
+exports.me = (req, res) => {
   res.jsonp(req.user || null);
 };
 
 /**
  * Find user by id
+ * @param {object} req - the request object
+ * @param {object} res - the response object
+ * @param {callback} next - a callback function
+ * @param {number} id - the ID of the user to be found
+ * @returns {func} Returns the found user
  */
-exports.user = function(req, res, next, id) {
+exports.user = (req, res, next, id) => {
   User
     .findOne({
       _id: id
     })
-    .exec(function(err, user) {
+    .exec((err, user) => {
```

Comment:
Great Job here, I can see you modified the code to use ES6 arrow function.

Reply:
Thanks for the feedback.